File: host/src/main/java/dk/dtu/domain/model/Board.java
package dk.dtu.domain.model;

/**
 * Represents a two-dimensional game board composed of tiles.
 *
 * @author William Pii Jæger
 */
public class Board {
    private final int width;
    private final int height;
    private final Tile[][] tiles;

    public Board(int width, int height, Tile[][] tiles) {
        this.width = width;
        this.height = height;
        this.tiles = tiles;
    }

    public int getWidth() {
        return width;
    }

    public int getHeight() {
        return height;
    }

    public Tile[][] getCells() {
        return tiles;
    }

    public boolean isInBounds(int x, int y) {
        return x >= 0 && y >= 0 && x < width && y < height;
    }

    public Tile getTile(int x, int y) { return tiles[x][y]; };

}

File: engine/src/model/board.ts
import { Tile } from "./tile.js";

/** Ported from dk.dtu.domain.model.Board. Tiles stored column-major: tiles[x][y]. */
export class Board {
  constructor(
    readonly width: number,
    readonly height: number,
    private readonly tiles: Tile[][],
  ) {}

  getWidth(): number {
    return this.width;
  }

  getHeight(): number {
    return this.height;
  }

  getCells(): Tile[][] {
    return this.tiles;
  }

  getTiles(): Tile[][] {
    return this.tiles;
  }

  isInBounds(x: number, y: number): boolean {
    return x >= 0 && y >= 0 && x < this.width && y < this.height;
  }

  getTile(x: number, y: number): Tile {
    return this.tiles[x][y];
  }
}
